src/main/com/yetanalytics/squuid.cljc

```diff
@@ -94,3 +95,20 @@
   [ts]
   (:squuid
    (u/make-squuid ts #uuid "00000000-0000-4FFF-8FFF-FFFFFFFFFFFF")))
+
+(s/fdef uuid->time
+  :args (s/cat :uuid ::squuid)
+  :ret ::timestamp)
+
+(defn uuid->time
+  "Convert a previously generated UUID to its corresponding timestamp."
+  [uuid]
+  #?(:clj
+     (-> (.getMostSignificantBits uuid)
```

Comment:
Side note, but I thought this line would've caused a reflection warning but I checked and it doesn't, so I guess we're good.